Sources/Conformance/conformance.pb.swift

```diff
@@ -531,50 +529,38 @@ extension Conformance_ConformanceRequest: SwiftProtobuf.Message, SwiftProtobuf._
     }
   }
 
-  func traverse<V: SwiftProtobuf.Visitor>(visitor: inout V) throws {
-    // The use of inline closures is to circumvent an issue where the compiler
-    // allocates stack space for every if/case branch local when no optimizations
-    // are enabled. https://github.com/apple/swift-protobuf/issues/1034 and
-    // https://github.com/apple/swift-protobuf/issues/1182
-    switch self.payload {
-    case .protobufPayload?: try {
-      guard case .protobufPayload(let v)? = self.payload else { preconditionFailure() }
-      try visitor.visitSingularBytesField(value: v, fieldNumber: 1)
-    }()
-    case .jsonPayload?: try {
-      guard case .jsonPayload(let v)? = self.payload else { preconditionFailure() }
-      try visitor.visitSingularStringField(value: v, fieldNumber: 2)
-    }()
-    default: break
-    }
-    if self.requestedOutputFormat != .unspecified {
-      try visitor.visitSingularEnumField(value: self.requestedOutputFormat, fieldNumber: 3)
-    }
-    if !self.messageType.isEmpty {
-      try visitor.visitSingularStringField(value: self.messageType, fieldNumber: 4)
-    }
-    if self.testCategory != .unspecifiedTest {
-      try visitor.visitSingularEnumField(value: self.testCategory, fieldNumber: 5)
-    }
-    try { if let v = self._jspbEncodingOptions {
-      try visitor.visitSingularMessageField(value: v, fieldNumber: 6)
-    } }()
-    switch self.payload {
-    case .jspbPayload?: try {
-      guard case .jspbPayload(let v)? = self.payload else { preconditionFailure() }
-      try visitor.visitSingularStringField(value: v, fieldNumber: 7)
-    }()
-    case .textPayload?: try {
-      guard case .textPayload(let v)? = self.payload else { preconditionFailure() }
-      try visitor.visitSingularStringField(value: v, fieldNumber: 8)
-    }()
-    default: break
-    }
-    if self.printUnknownFields != false {
-      try visitor.visitSingularBoolField(value: self.printUnknownFields, fieldNumber: 9)
-    }
-    try unknownFields.traverse(visitor: &visitor)
-  }
+  static let _fields: [Field<Self>] = [
+    .oneOf({ $0.payload }) {
+      switch $0 {
+      case .protobufPayload:
+        return _oneOfField_protobufPayload
+      case .jsonPayload:
+        return _oneOfField_jsonPayload
+      default:
+        return nil
+      }
+    },
+    .singularEnum({ $0.requestedOutputFormat }, fieldNumber: 3, defaultValue: .unspecified),
+    .singularString({ $0.messageType }, fieldNumber: 4),
+    .singularEnum({ $0.testCategory }, fieldNumber: 5, defaultValue: .unspecifiedTest),
+    .singularMessage({ $0.jspbEncodingOptions }, fieldNumber: 6, isUnset: { $0._jspbEncodingOptions == nil }),
+    .oneOf({ $0.payload }) {
+      switch $0 {
+      case .jspbPayload:
+        return _oneOfField_jspbPayload
+      case .textPayload:
+        return _oneOfField_textPayload
+      default:
+        return nil
+      }
+    },
+    .singularBool({ $0.printUnknownFields }, fieldNumber: 9),
```

Comment:
> Would it help to combine this with the _protobuf_nameMap information ? Superficially, it seems like it would only require adding name information to these items

Yes absolutely I think this design allows us to remove the `_protobuf_nameMap` and probably speed up json encoding as a result since we can get rid of the dictionary lookup. I didn't include it in this PR to keep it focused but I think that absolutely makes sense.

We could have `_fields` be an [OrderedDictionary](https://github.com/apple/swift-collections/blob/main/Documentation/OrderedDictionary.md) where the keys are the field numbers allow it to completely remove the need for `_protobuf_nameMap`. Allowing us to traverse fields in an ordered way while still allowing us to look up a field name from a given field number for decoding purposes.

Reply:
Hmmm....   So many trade-offs!   OrderedDictionary would have additional runtime memory overhead, of course.

I know the compiler folks are working on techniques to make the compiler better at coding constant arrays in TEXT at compile-time rather than building them in dirty memory at runtime.  But to take advantage of that, we'd need to simplify that array construction a lot.  That's somewhat speculative, but would suggest that it would be worth building the Fields array as a simple list of constant enum cases, eliminating as much indirection as possible.

Reply:
The lookup is just for parsing, right?  With a low number of fields, walking a list is likely cost the the performance of searching a dictionary. And we might even be able to tweak things to optimize for an ordered input also.  i.e. - a table driven parse might be something to also look at doing, reducing code size even more.

Reply:
> ... the compiler was deciding to create copies of the enum case and its content during the switch ...

Ah.  Yeah, I know that problem well. 😞